Added support for PMA_PASSWORD_FILE

Mandy91 opened the pull request:
Documentation said there is support for, but there was no code to implement this

williamdes commented:
@davidmwhynot is this change needed ?
#232

davidmwhynot commented:
Based on [this](https://github.com/phpmyadmin/docker/issues/108#issuecomment-519323193) I would say no.

davidmwhynot commented on docker-entrypoint.sh:
```diff
@@ -47,4 +47,8 @@ EOT
 fi
 
+if [ -n "$PMA_PASSWORD_FILE" ]; then
+	export PMA_PASSWORD="$(cat $PMA_PASSWORD_FILE)"
```
This implementation bypasses one of the main advantages of using docker secrets: sensitive information is not stored directly in the an environment variable. Instead, it is read from a secret file at runtime by application code. Nothing sensitive should ever be stored directly in an environment variable, as is being done here.

davidmwhynot commented:
> Based on [this](https://github.com/phpmyadmin/docker/issues/108#issuecomment-519323193) I would say no.

After doing some digging through the repository, I am unable to find an implementation for actually reading secrets from a file path specified by an environment variable. So not really sure where [this](https://github.com/phpmyadmin/docker/issues/108#issuecomment-519323193) information came from.

williamdes commented:
@J0WI can I please have your review on this one ?

J0WI commented:
Not sure why #232 have been merged, this functionality was never implemented.
@Mandy91 there is a great and roust implementation in #206 (taken from [here](https://github.com/docker-library/wordpress/blob/master/docker-entrypoint.sh#L4-L24)). Would you like to pick this up again?

williamdes commented:
> Not sure why #232 have been merged, this functionality was never implemented.

I trusted the comment and then merged the PR..

Mandy91 commented:
@J0WI Sure no problem. Would the following environment variables suffice to implement the password file for? To me it doesn't make much sense to add it for the username environment variable etc.
- MYSQL_ROOT_PASSWORD
- MYSQL_PASSWORD
- PMA_PASSWORD

J0WI commented:
> To me it doesn't make much sense to add it for the username environment variable etc.

Sure
cc @williamdes

williamdes commented:
I agree

williamdes commented on testing/conftest.py:
```diff
@@ -0,0 +1,28 @@
+    
+@pytest.fixture
+def sqlfile(request):
+    return request.config.getoption("--sqlfile")
```
can you add a final line ?

williamdes commented on testing/phpmyadmin_test.py:
```diff
@@ -20,12 +20,14 @@ def test_phpmyadmin(url, username, password, server=None, sqlfile=None):
     # Ignore robots.txt
     br.set_handle_robots(False)
 
+
```
can you remove the new line here please

Mandy91 commented:
@williamdes Can we somehow trigger Travis to see if the tests which I have created work properly?

williamdes commented on docker-entrypoint.sh:
```diff
@@ -47,4 +47,17 @@ EOT
+    local env_var="${1}"
+    local env_var_file="${env_var}_FILE"
+
+    if [[ -n "${!env_var_file}" ]]; then
```
can you add a comment because it is unclear what this line does

williamdes commented on testing/run-test-locally.sh:
```diff
@@ -0,0 +1,32 @@
+docker kill phpmyadmin
+
+set -e
+docker run --rm -d -p 3300:3306 -e MYSQL_ROOT_PASSWORD=my-secret-pw --init --hostname mariadb --name mariadb --network phpmyadmin --tmpfs /var/lib/mysql:rw,noexec,nosuid,size=300m mariadb:10.3
```
coud you use some ENV variables that have default values like `${DOCKER_MYSQL_PORT:-3306}` ?

williamdes commented:
> @williamdes Can we somehow trigger Travis to see if the tests which I have created work properly?

I asked @ibennetch

williamdes commented:
Thank you for your work ! :100:

williamdes commented on docker-compose.testing.yml:
```diff
@@ -1,26 +1,27 @@
+      - MYSQL_ROOT_PASSWORD=${TESTSUITE_PASSWORD:-my-secret-pw}
     volumes:
       - ./testing/testing.cnf:/etc/mysql/conf.d/testing.cnf:ro
+    tmpfs: 
```
will have issues on MacOS but I agree to have that line

williamdes commented:
@Mandy91 did you discover why we have randomly failing E2E tests here ?

Mandy91 commented:
> @Mandy91 did you discover why we have randomly failing E2E tests here ?

Yes, initially I removed the variable FILENAME in test-docker.sh, I added it back and now it is working again :+1:

eriksenonline commented:
Is it possible to add support for PMA_HOSTS_FILE and PMA_HOST_FILE to ?

williamdes commented:
@Mandy91 I rebased your work

williamdes commented:
@Mandy91 I rebased and squashed your work

williamdes commented:
@Mandy91 If you want to edit your commit to use your account like previous ones please do so because I will merge in less than one hour :)

williamdes commented:
`From: Mandy <mandy@Mandys-MacBook-Pro.local>`

williamdes commented:
Merged, thank you for your contribution @Mandy91 !
You added tests and I love it !

I added b295aa6bf722a5e95d4b4b185e367e9a2d9c5916 after merging
